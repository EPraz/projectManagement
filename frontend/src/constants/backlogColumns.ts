// Actualiza tu interfaz para incluir una propiedad de orden
export interface BacklogColumnsProps {
  key: string;
  label: string;
  order: number; // Añadimos una propiedad de orden explícita
  width?: string; // Opcional: ancho personalizado
}

// Actualiza tus columnas con la propiedad de orden
export const backlogColumns: BacklogColumnsProps[] = [
  { key: "order", label: "Order", order: 1, width: "10%" },
  { key: "id", label: "ID", order: 2, width: "10%" },
  { key: "state", label: "State", order: 3, width: "15%" },
  { key: "title", label: "Title", order: 4 },
  { key: "type", label: "Type", order: 5 },
  { key: "priority", label: "Priority", order: 6, width: "10%" },
  { key: "assignedTo", label: "Assigned To", order: 7 },
  { key: "sprint", label: "Sprint", order: 8 },
  { key: "dueDate", label: "Due Date", order: 9 },
  { key: "estimatedHours", label: "Est. Hours", order: 10, width: "8%" },
  { key: "remainingHours", label: "Rem. Hours", order: 11, width: "8%" },
  { key: "isBlocked", label: "Blocking", order: 12 },
  { key: "tags", label: "Tags", order: 13 },
  { key: "actions", label: "Actions", order: 14, width: "5%" },
];

// Columnas inicialmente visibles
export const initialBacklogColumns: string[] = [
  "order",
  "id",
  "state",
  "title",
  "priority",
  "tags",
  "actions",
];
