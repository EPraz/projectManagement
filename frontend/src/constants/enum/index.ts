export * from "./Role";
export * from "./TicketPriority";
export * from "./TicketType";
