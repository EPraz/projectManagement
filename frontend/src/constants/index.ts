export * from "./taskStatus";
export * from "./ticketStatus";
export * from "./enum";
export * from "./menuItems";
export * from "./tabPanels";
export * from "./backlogColumns";
export * from "./apiConstant";
