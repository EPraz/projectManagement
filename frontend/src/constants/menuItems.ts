import DashboardIcon from "@mui/icons-material/Dashboard";
import ViewKanbanIcon from "@mui/icons-material/ViewKanban";
import AssessmentIcon from "@mui/icons-material/Assessment";
import CategoryIcon from "@mui/icons-material/Category";
import PeopleIcon from "@mui/icons-material/People";
import SettingsIcon from "@mui/icons-material/Settings";

export interface MenuItem {
  text: string;
  icon: React.ElementType;
  path: string;
  disabled?: boolean;
}

export const MENU_ITEMS: MenuItem[] = [
  {
    text: "Overview",
    icon: DashboardIcon,
    path: "overview",
  },
  {
    text: "Board",
    icon: ViewKanbanIcon,
    path: "board",
  },
  {
    text: "Retrospective",
    icon: AssessmentIcon,
    path: "retrospective",
  },
  {
    text: "Epics & Features",
    icon: CategoryIcon,
    path: "epics",
    disabled: true,
  },

  {
    text: "Team Members",
    icon: PeopleIcon,
    path: "teammembers",
  },
  {
    text: "Settings",
    icon: SettingsIcon,
    path: "settings",
  },
];

export interface MainTab {
  text: string;
  path: string;
}

export const MAIN_TABS: MainTab[] = [
  {
    text: "Overview",
    path: "overview",
  },
  {
    text: "Board",
    path: "board",
  },
  {
    text: "Members",
    path: "members",
  },
  {
    text: "Files",
    path: "files",
  },
  {
    text: "Timeline",
    path: "timeline",
  },
  {
    text: "Reports",
    path: "reports",
  },
];

export interface Project {
  name: string;
  count: number;
  color: string;
}

export const PROJECTS: Project[] = [
  { name: "ShieldSafe", count: 5, color: "#1E8E3E" },
  { name: "Quantum", count: 2, color: "#1A73E8" },
  { name: "AeroSquare", count: 7, color: "#F9AB00" },
];

export const cardVariants = {
  hidden: { opacity: 0, y: 20 },
  visible: { opacity: 1, y: 0 },
};
