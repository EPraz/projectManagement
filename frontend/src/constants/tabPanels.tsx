import ViewKanbanIcon from "@mui/icons-material/ViewKanban";
import FormatListBulletedIcon from "@mui/icons-material/FormatListBulleted";
import GroupIcon from "@mui/icons-material/Group";
import FlagIcon from "@mui/icons-material/Flag";

import { Taskboard } from "../pages";
import BacklogBoard from "../pages/board/backlogBoard/BacklogBoard";
import CapacityBoard from "../pages/board/capacityBoard/CapacityBoard";
import GoalBoard from "../pages/board/goalBoard/GoalBoard";

interface TabPanel {
  label: string;
  icon: JSX.Element;
  component: () => JSX.Element;
}

export const TabPanels: TabPanel[] = [
  { label: "Board", icon: <ViewKanbanIcon />, component: () => <Taskboard /> },
  {
    label: "Backlog",
    icon: <FormatListBulletedIcon />,
    component: () => <BacklogBoard />,
  },
  {
    label: "Capacity",
    icon: <GroupIcon />,
    component: () => <CapacityBoard />,
  },
  { label: "Goals", icon: <FlagIcon />, component: () => <GoalBoard /> },
];
