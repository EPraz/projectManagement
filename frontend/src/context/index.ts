export * from "./sprintContext";
export * from "./apiContext";
export * from "./projectContext";
export * from './snackbarContext'