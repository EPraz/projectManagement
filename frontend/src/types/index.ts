export * from "./contextProps";
export * from "./models";
export * from "./materialUiExtensions";
export * from "./components";
export * from "./FilterProps";
