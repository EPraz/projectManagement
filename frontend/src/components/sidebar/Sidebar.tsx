import React, { useMemo, useState } from "react";
import { useNavigate } from "react-router-dom";
import { List, Avatar, Box } from "@mui/material";
import { MENU_ITEMS } from "../../constants";
import { ThemeToggleModal } from "../themeToggleModal/ThemeToggleModal";
import type { SidebarProps } from "../../types";
import KeyboardDoubleArrowLeftOutlinedIcon from "@mui/icons-material/KeyboardDoubleArrowLeftOutlined";
import KeyboardDoubleArrowRightOutlinedIcon from "@mui/icons-material/KeyboardDoubleArrowRightOutlined";
import {
  CollapseButton,
  Logo,
  LogoText,
  SidebarPlaceholder,
  StyledDrawer,
  StyledListItemButton,
  StyledListItemIcon,
  StyledListItemText,
  UserBox,
  UserInfo,
  UserName,
  UserSection,
  WorkspaceText,
} from "./Sidebar.styles";

const Sidebar: React.FC<SidebarProps> = ({
  project,
  sidebarOpen,
  setSidebarOpen,
}) => {
  const [themeModalOpen, setThemeModalOpen] = useState(false);

  const navigate = useNavigate();

  const menuItems = useMemo(
    () =>
      MENU_ITEMS.map((item) => ({
        text: item.text,
        icon: <item.icon />,
        path: `/projects/${project.id}/${item.path}`,
        disabled: item.disabled,
      })),
    [MENU_ITEMS]
  );

  const toggleCollapse = () => {
    const x = !sidebarOpen;
    setSidebarOpen(x);
    localStorage.setItem("sidebarOpen", x.toString());
  };

  return (
    <>
      <SidebarPlaceholder $IsOpen={sidebarOpen} />
      <StyledDrawer variant="permanent" $IsOpen={sidebarOpen}>
        <Logo>
          <Box sx={{ display: "flex", alignItems: "center" }}>
            <svg
              width="24"
              height="24"
              viewBox="0 0 24 24"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM12 20C7.59 20 4 16.41 4 12C4 7.59 7.59 4 12 4C16.41 4 20 7.59 20 12C20 16.41 16.41 20 12 20Z"
                fill="#1E8E3E"
              />
              <path
                d="M12 11C13.1046 11 14 10.1046 14 9C14 7.89543 13.1046 7 12 7C10.8954 7 10 7.89543 10 9C10 10.1046 10.8954 11 12 11Z"
                fill="#1E8E3E"
              />
              <path
                d="M12 13C10.8954 13 10 13.8954 10 15C10 16.1046 10.8954 17 12 17C13.1046 17 14 16.1046 14 15C14 13.8954 13.1046 13 12 13Z"
                fill="#1E8E3E"
              />
              <path
                d="M9 12C9 13.1046 8.10457 14 7 14C5.89543 14 5 13.1046 5 12C5 10.8954 5.89543 10 7 10C8.10457 10 9 10.8954 9 12Z"
                fill="#1E8E3E"
              />
              <path
                d="M17 12C17 13.1046 16.1046 14 15 14C13.8954 14 13 13.1046 13 12C13 10.8954 13.8954 10 15 10C16.1046 10 17 10.8954 17 12Z"
                fill="#1E8E3E"
              />
            </svg>
            <LogoText $IsOpen={sidebarOpen}>{project.title}</LogoText>
          </Box>
        </Logo>

        <List sx={{ padding: "8px 0" }}>
          {menuItems.map(({ text, icon, path, disabled }) => (
            <StyledListItemButton
              key={text}
              onClick={() => navigate(path)}
              className={
                path === `/projects/${project.id}/dashboard` ? "active" : ""
              }
              sx={{ justifyContent: sidebarOpen ? "center" : "flex-start" }}
              disabled={disabled}
            >
              <StyledListItemIcon sx={{ minWidth: sidebarOpen ? 0 : 40 }}>
                {icon}
              </StyledListItemIcon>
              <StyledListItemText primary={text} $IsOpen={sidebarOpen} />
            </StyledListItemButton>
          ))}
        </List>

        <Box sx={{ flexGrow: 1 }} />

        <UserSection $IsOpen={sidebarOpen}>
          <UserBox $IsOpen={sidebarOpen}>
            <Avatar
              alt="User"
              src="/placeholder.svg?height=40&width=40"
              sx={{ width: 36, height: 36 }}
            >
              P
            </Avatar>
            <UserInfo $IsOpen={sidebarOpen}>
              <UserName>Phenomenon</UserName>
              <WorkspaceText>Workspace</WorkspaceText>
            </UserInfo>
          </UserBox>
          {sidebarOpen ? (
            <CollapseButton onClick={toggleCollapse}>
              <KeyboardDoubleArrowRightOutlinedIcon />
            </CollapseButton>
          ) : (
            <>
              {/* <ThemeToggle onClick={() => setThemeModalOpen(true)}>
                <DarkModeIcon />
                </ThemeToggle> */}
              <CollapseButton onClick={toggleCollapse}>
                <KeyboardDoubleArrowLeftOutlinedIcon />
              </CollapseButton>
            </>
          )}
        </UserSection>
      </StyledDrawer>

      <ThemeToggleModal
        open={themeModalOpen}
        onClose={() => setThemeModalOpen(false)}
      />
    </>
  );
};

export default Sidebar;
