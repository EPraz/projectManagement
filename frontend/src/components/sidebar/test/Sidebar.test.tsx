// Sidebar.test.tsx
import { render, screen, fireEvent } from "@testing-library/react";
import Sidebar from "../Sidebar";
import { MemoryRouter } from "react-router-dom";
import { Project } from "../../../types";
import { MENU_ITEMS } from "../../../constants";
import { createAppTheme } from "../../../utils";
import { ThemeProvider as MuiThemeProvider } from "@mui/material/styles";
import { ThemeProvider as CustomThemeProvider } from "../../../context"; // Asegúrate de usar el provider que creaste en tu contexto
import { ThemeProvider as StyledComponentsThemeProvider } from "styled-components";

// Para poder testear la navegación, mockeamos useNavigate
const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => {
  const actual = jest.requireActual("react-router-dom");
  return {
    ...actual,
    useNavigate: () => mockNavigate,
  };
});

describe("Sidebar Component", () => {
  const project: Project = {
    id: "123",
    title: "Test Project",
    createdBy: "example",
    createdAt: "zxcv",
    updatedAt: "asdf",
    epics: [],
    tickets: [],
    users: [],
    sprints: [],
    ticketStatuses: [],
    taskStatuses: [],
    epicStatuses: [],
    featureStatuses: [],
    type: "hello",
    productManagers: [],
  };
  const setSidebarOpenMock = jest.fn();

  // Para los tests, usamos el modo "light" (o el que necesites)
  const mode = "light";
  const theme = createAppTheme(mode);

  const renderSidebar = (sidebarOpen: boolean) =>
    render(
      <CustomThemeProvider>
        <MuiThemeProvider theme={theme}>
          <StyledComponentsThemeProvider theme={theme}>
            <MemoryRouter>
              <Sidebar
                project={{ id: "123", title: "Test Project" } as any}
                sidebarOpen={sidebarOpen}
                setSidebarOpen={setSidebarOpenMock}
              />
            </MemoryRouter>
          </StyledComponentsThemeProvider>
        </MuiThemeProvider>
      </CustomThemeProvider>
    );

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("match snapshot", () => {
    const { asFragment } = renderSidebar(true);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders logo with project title when sidebar is open", () => {
    renderSidebar(true);
    expect(screen.getByText("Test Project")).toBeInTheDocument();
  });

  it("renders menu items based on MENU_ITEMS and navigates on click", () => {
    renderSidebar(true);
    MENU_ITEMS.forEach((item) => {
      expect(screen.getByText(item.text)).toBeInTheDocument();
    });
    const firstItem = screen.getByText(MENU_ITEMS[0].text);
    fireEvent.click(firstItem);
    expect(mockNavigate).toHaveBeenCalledWith(
      `/projects/${project.id}/${MENU_ITEMS[0].path}`
    );
  });

  it("calls setSidebarOpen on collapse button click", () => {
    renderSidebar(true);
    const collapseButton = screen.getByTestId("collapse-button");
    fireEvent.click(collapseButton);
    expect(setSidebarOpenMock).toHaveBeenCalledWith(false);
  });
});
