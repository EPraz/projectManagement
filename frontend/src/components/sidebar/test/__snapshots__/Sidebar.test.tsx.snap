// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1mp49yy"
  />
  <div
    class="MuiDrawer-root MuiDrawer-anchorLeft MuiDrawer-docked css-k6vpck-MuiDrawer-docked"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-m25uzs-MuiPaper-root-MuiDrawer-paper"
      style="--Paper-shadow: none;"
    >
      <div
        class="css-1t09gyw"
      >
        <div
          class="MuiBox-root css-70qvj9"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM12 20C7.59 20 4 16.41 4 12C4 7.59 7.59 4 12 4C16.41 4 20 7.59 20 12C20 16.41 16.41 20 12 20Z"
              fill="#1E8E3E"
            />
            <path
              d="M12 11C13.1046 11 14 10.1046 14 9C14 7.89543 13.1046 7 12 7C10.8954 7 10 7.89543 10 9C10 10.1046 10.8954 11 12 11Z"
              fill="#1E8E3E"
            />
            <path
              d="M12 13C10.8954 13 10 13.8954 10 15C10 16.1046 10.8954 17 12 17C13.1046 17 14 16.1046 14 15C14 13.8954 13.1046 13 12 13Z"
              fill="#1E8E3E"
            />
            <path
              d="M9 12C9 13.1046 8.10457 14 7 14C5.89543 14 5 13.1046 5 12C5 10.8954 5.89543 10 7 10C8.10457 10 9 10.8954 9 12Z"
              fill="#1E8E3E"
            />
            <path
              d="M17 12C17 13.1046 16.1046 14 15 14C13.8954 14 13 13.1046 13 12C13 10.8954 13.8954 10 15 10C16.1046 10 17 10.8954 17 12Z"
              fill="#1E8E3E"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-16w7but-MuiTypography-root"
          >
            Test Project
          </p>
        </div>
      </div>
      <ul
        class="MuiList-root MuiList-padding css-1qeyll5-MuiList-root"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-wwerzn-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root css-13iiu07-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="DashboardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 13h8V3H3zm0 8h8v-6H3zm10 0h8V11h-8zm0-18v6h8V3z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-i4gb3i-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-9zsfkx-MuiTypography-root"
            >
              Overview
            </span>
          </div>
        </div>
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-wwerzn-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root css-13iiu07-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="ViewKanbanIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2M9 17H7V7h2zm4-5h-2V7h2zm4 3h-2V7h2z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-i4gb3i-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-9zsfkx-MuiTypography-root"
            >
              Board
            </span>
          </div>
        </div>
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-wwerzn-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root css-13iiu07-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="AssessmentIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2M9 17H7v-7h2zm4 0h-2V7h2zm4 0h-2v-4h2z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-i4gb3i-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-9zsfkx-MuiTypography-root"
            >
              Retrospective
            </span>
          </div>
        </div>
        <div
          aria-disabled="true"
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters Mui-disabled Mui-disabled MuiListItemButton-root MuiListItemButton-gutters Mui-disabled css-wwerzn-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="-1"
        >
          <div
            class="MuiListItemIcon-root css-13iiu07-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="CategoryIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m12 2-5.5 9h11z"
              />
              <circle
                cx="17.5"
                cy="17.5"
                r="4.5"
              />
              <path
                d="M3 13.5h8v8H3z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-i4gb3i-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-9zsfkx-MuiTypography-root"
            >
              Epics & Features
            </span>
          </div>
        </div>
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-wwerzn-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root css-13iiu07-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="PeopleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5s-3 1.34-3 3 1.34 3 3 3m-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5 5 6.34 5 8s1.34 3 3 3m0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5m8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-i4gb3i-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-9zsfkx-MuiTypography-root"
            >
              Team Members
            </span>
          </div>
        </div>
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-wwerzn-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root css-13iiu07-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="SettingsIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-i4gb3i-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-9zsfkx-MuiTypography-root"
            >
              Settings
            </span>
          </div>
        </div>
      </ul>
      <div
        class="MuiBox-root css-i9gxme"
      />
      <div
        class="MuiBox-root css-i86ng8"
      >
        <div
          class="MuiBox-root css-100bdpa"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-177uw2a-MuiAvatar-root"
          >
            <img
              alt="User"
              class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
              src="/placeholder.svg?height=40&width=40"
            />
          </div>
          <div
            class="MuiBox-root css-19nagu4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-17m4wal-MuiTypography-root"
            >
              Phenomenon
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-dp3r4l-MuiTypography-root"
            >
              Workspace
            </p>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1od977l-MuiButtonBase-root-MuiIconButton-root"
          data-testid="collapse-button"
          id=":r0:"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="KeyboardDoubleArrowRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6.41 6 5 7.41 9.58 12 5 16.59 6.41 18l6-6z"
            />
            <path
              d="m13 6-1.41 1.41L16.17 12l-4.58 4.59L13 18l6-6z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
